src/Compilers/CSharp/Portable/Lowering/LambdaRewriter/LambdaRewriter.LocalFunctionReferenceRewriter.cs

```diff
@@ -82,10 +82,8 @@ public override BoundNode VisitDelegateCreationExpression(BoundDelegateCreationE
                         out method,
                         ref arguments);
 
-                    var newType = _lambdaRewriter.VisitType(node.Type);
```

Comment:
I just inlined that call. It's still necessary. I do like the assert idea though, I will add that.